source/git-api.js

```diff
@@ -692,23 +692,33 @@ exports.registerApi = (env) => {
     });
     app.post(`${exports.pathPrefix}/testing/createfile`, ensureAuthenticated, (req, res) => {
       const content = req.body.content ? req.body.content : (`test content\n${Math.random()}\n`);
-      fs.writeFile(req.body.file, content, () => res.json({}));
+      fs.writeFileAsync(req.body.file, content)
+        .then(() => res.json({}))
+        .then(emitWorkingTreeChanged.bind(null, req.body.path));
     });
     app.post(`${exports.pathPrefix}/testing/changefile`, ensureAuthenticated, (req, res) => {
       const content = req.body.content ? req.body.content : (`test content\n${Math.random()}\n`);
-      fs.writeFile(req.body.file, content, () => res.json({}));
+      fs.writeFileAsync(req.body.file, content)
+        .then(() => res.json({}))
+        .then(emitWorkingTreeChanged.bind(null, req.body.path));
     });
      app.post(`${exports.pathPrefix}/testing/createimagefile`, ensureAuthenticated, (req, res) => {
-      fs.writeFile(req.body.file, 'png', { encoding: 'binary' }, () => res.json({}));
+      fs.writeFileAsync(req.body.file, 'png', { encoding: 'binary' })
+        .then(() => res.json({}))
+        .then(emitWorkingTreeChanged.bind(null, req.body.path));
     });
     app.post(`${exports.pathPrefix}/testing/changeimagefile`, ensureAuthenticated, (req, res) => {
-      fs.writeFile(req.body.file, 'png ~~', { encoding: 'binary' }, () => res.json({}));
+      fs.writeFileAsync(req.body.file, 'png ~~', { encoding: 'binary' })
+        .then(() => res.json({}))
+        .then(emitWorkingTreeChanged.bind(null, req.body.path));
     });
     app.post(`${exports.pathPrefix}/testing/removefile`, ensureAuthenticated, (req, res) => {
-      fs.unlink(req.body.file, () => res.json({}));
+      fs.unlinkAsync(req.body.file)
+        .then(() => res.json({}))
+        .then(emitWorkingTreeChanged.bind(null, req.body.path));
     });
     app.post(`${exports.pathPrefix}/testing/git`, ensureAuthenticated, (req, res) => {
-      jsonResultOrFailProm(res, gitPromise(req.body.command, req.body.repo));
+      jsonResultOrFailProm(res, gitPromise(req.body.command, req.body.path));
```

Comment:
This was the only case where the repro path variable name was `repo`. I just changed it to `path` to be consistent with the rest.